src/features/layers_in_area/atoms/areaLayersLegendsAndSources.ts

```diff
@@ -63,6 +67,16 @@ export const areaLayersLegendsAndSources = createAtom(
         return acc;
       }, new Map<string, LayerInAreaDetails>());
 
+      // apply cached layers if any are already stored for current eventId
+      const focusedGeometry = get('focusedGeometryAtom');
+      const eventId = getEvendId(focusedGeometry);
+      if (eventId) {
+        const cachedLayers = getUnlistedState(areaLayersDetailsResourceAtomCache).get(
+          eventId,
+        );
+        cachedLayers?.forEach((layer) => layersDetailsData.set(layer.id, layer));
```

Comment:
The idea was that each update of the resource atom only adds new data to layersSourcesAtom and layersLegendsAtom
(83 and 97 lines of this file). So the cache in this case is only needed to compose the new request.

I didn't take into account that they are deleted somewhere else.

I think we need a test for this behavior, this bug has been on the production for some time now because it's hard to catch it by hand